modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/snapshot/WeightedSamplingSuite.scala

```diff
@@ -0,0 +1,62 @@
+package org.tessellation.sdk.infrastructure.snapshot
+
+import cats.syntax.applicative._
+
+import org.tessellation.sdk.infrastructure.snapshot.WeightedSampling.{ProbabilityRefinement, WeightRefined}
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.cats.refTypeShow
+import eu.timepit.refined.refineV
+import org.scalacheck.Gen
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object WeightedSamplingSuite extends SimpleIOSuite with Checkers {
+
+  val distribution: List[(String, WeightRefined)] = List(
+    "a" -> 0.1,
+    "b" -> 0.5,
+    "c" -> 0.3
+  )
+  val distributionSum: Double = distribution.foldLeft(0.0)(_ + _._2)
+
+  test("select a") {
```

Comment:
can you describe here the logic not the effect? why it should choose a?

Reply:
Made the test names more descriptive.